src/osp/Resource/Package.h

```diff
@@ -61,12 +61,26 @@ namespace osp
 
 // supported resources
 
+// Just a string, but typedef'd to indicate that it represents a prefix
+typedef std::string resprefix_t;
 
-class Package
-{
-
+// Stores a split path as <prefix, rest of path>
+typedef struct Path { resprefix_t prefix; std::string identifier; } path_t;
```

Comment:
There's no reason to make this a typedef at all. Referencing it by the name of the struct type is perfectly fine.

Reply:
Also, man, that's a lot of string copying. You sure? I feel like what you want here is actually two std::string_views and then a single string object that owns the storage (Careful with copy / move constructors though).

Or maybe even more accurately, you might want something like an std::string_range (doesn't exist, far as I know) that's basically just two size_t's that indicate the range of indicies in some std::string, applied as needed to get back an std::string_view.

Reply:
Very C-style indeed. There's many ways to go upon doing this, depending on how this struct will be used. Will this struct own a string or is it just a return type for decompose_path?

Reply:
Can't remember how it ended up this way, I suspect I originally typedef'd a `std::pair` or something and then forgot to get rid of it when I changed to a struct.

Reply:
Looks like string_view will work pretty much everywhere, I will either just change everything to string_view or make an equivalent typedef for the resprefix, depending on whether cap thinks it's reasonable to use the typedef in the first place.